lib/contentstack/api.rb

```diff
@@ -75,12 +95,30 @@ def self.send_request(path, q=nil)
         "api_key" =>  @api_key,
         "access_token"=>  @access_token,
         "user_agent"=> "ruby-sdk/#{Contentstack::VERSION}",
-        "x-user-agent" => "ruby-sdk/#{Contentstack::VERSION}"
+        "x-user-agent" => "ruby-sdk/#{Contentstack::VERSION}",
+        "read_timeout" => @timeout
       }
       if !@branch.nil? && !@branch.empty?
         params["branch"] = @branch
       end
-      ActiveSupport::JSON.decode(URI.open("#{@host}#{@api_version}#{path}#{query}", params).read)
+      if @proxy_details.empty?
+
+        ActiveSupport::JSON.decode(URI.open("#{@host}#{@api_version}#{path}#{query}", params).read)
+
+      elsif @proxy_details.present? && @proxy_details[:url].present? && @proxy_details[:port].present? && @proxy_details[:username].present? && @proxy_details[:password].present?
+          
+        proxy_uri = URI.parse("http://#{@proxy_details[:url]}:#{@proxy_details[:port]}/")
+        proxy_username = @proxy_details[:username]
+        proxy_password = @proxy_details[:password]
+       
+        ActiveSupport::JSON.decode(URI.open("#{@host}#{@api_version}#{path}#{query}", :proxy_http_basic_authentication => [proxy_uri, proxy_username, proxy_password], "api_key" =>  @api_key, "access_token"=>  @access_token, "user_agent"=> "ruby-sdk/#{Contentstack::VERSION}", "x-user-agent" => "ruby-sdk/#{Contentstack::VERSION}", "read_timeout" => @timeout).read)
```

Comment:
@sunil-lakshman  Branch Header is missing in this request. We need to check if branch is passed in Client initialisation if its present we should add that as header in request.